### ISSUE TITLE
The nested test suites are executed repeatedly if no test suite is specified

### ISSUE THREAD

**Issue by gundamew** [sebastianbergmann/phpunit#5670]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 10.5.7
| PHP version         | 8.2.14
| Installation Method | Composer

#### Summary

The nested test suites would be executed repeatedly if the user doesn't specify the test suite in PHPUnit 10.
In my case, it can be resolved with the `defaultTestSuite` attribute specified.

The test file structure:

```text
tests/unit
├─ FooTest.php
└─ BarTest.php

tests/integration
└─ BazTest.php
```

The test suite configuration:

```xml
<testsuites>
  <testsuite name="default">
    <directory>tests</directory>
  </testsuite>
  <testsuite name="unit">
    <directory>tests/unit</directory>
  </testsuite>
  <testsuite name="integration">
    <directory>tests/integration</directory>
  </testsuite>
</testsuites>
```

> Please do not report an issue for API changes in internal code (see https://phpunit.de/backward-compatibility.html#internal-code).

It's might about [internal code](https://github.com/sebastianbergmann/phpunit/blob/a6319b7dd20318e2578031ef847492a9c1cd9af8/src/Runner/TestSuiteLoader.php#L26). But I didn't see any information about the behavior. This issue is more like a guide for users with similar problems. So, I still choose to post it.

Please let me know if this is inappropriate or if I can find the information on how PHPUnit handles the test suites somewhere.

The senior developer of my organization, [Albert Wang](https://tw.linkedin.com/in/albert-wang-59357095), helped me to verify the issue. Thanks to him.

#### Current behavior

6 tests are executed. (Please see the sample repository below)

#### How to reproduce

Here is the sample repository: https://github.com/gundamew/phpunit10-nested-testsuite-issue

There are 2 unit tests and 1 integration test.

1. Clone it.
2. Switch to `phpunit9` or `phpunit10` branch.
3. Run `composer install`.
4. Run `./vendor/bin/phpunit -c ./phpunit.xml`.

#### Expected behavior

3 tests are executed.

**Comment by sebastianbergmann** [comment]
The behaviour of PHPUnit 9 was "by accident". The configuration of (nested) test suites assumes that there is no overlap between two test suites, meaning it is unexpected that the same test is part of multiple test suites.

IMO, the correct fix here would be to error out when the same test source file is found due to more than one `<testsuite>` element.

**Comment by sebastianbergmann** [comment]
Implemented in 57e07eef9de8e016be8da0f53a35d5df1fa72ebd for PHPUnit 11.
